### PR TITLE
WIP: user profile with history events

### LINKED ISSUE #2550: Add historical activity for particular user profile -- timeline
Start easy adding events for a particular address to show which item has address interacted.

Probably to particular whale directory #1165  related would be nice to estimate the daytime activity of particular address when the address has doing the on-chain activity.

Could be extended for the History component at profile? Like done https://github.com/kodadot/nft-gallery/issues/2346

💡 Open.
It would work as history sales feed for collections yet it would be on top of profile.

Inspired by nansen Alex.

### Ref
- inspired how block explorer works https://kusama.subscan.io/account/G29NScLSew5zqwmJAPupvJWDCDkpxKUhDnMeVdD2BBcnHar

### PR DESCRIPTION
**Pull request by MSghais**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [X] Feature
- [ ] Refactoring

### What's new?

- [X] PR closes #2550
- [X] Create a history event of one User Profil Detail selected. Use the history component with pagination.

### Before submitting Pull Request, please make sure:

- [X] My contribution builds **clean without any errors or warnings**
- [X] I've merged recent default branch -- **main** and I've no conflicts
- [X] I've tried to respect high code quality standards
- [X] I've didn't break any original functionality
- [X] I've posted a screenshot of demonstrated change in this PR

### Optional

- [] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [X] I've tested PR on mobile and everything seems works
- [X] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [X] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EFuhGSJpetTgoZ2dfWxwirxADBWgAzWmVfJmTCauZuPxz6z

### Community participation

- [X] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [X] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

Koda_work

### CODE REVIEW

**Comment by yangwao** [comment]
Hey, when I clicked on own profile I should land me on my KSM ss58 address

`/rmrk/u/CuHWHNcBt3ASMVSJmcJyiBWGxxiWLyjYoYbGjfhL4ovoeSd`
but at your PR I've landed on
`https://deploy-preview-2973--koda-nuxt.netlify.app/rmrk/u/5CPfqy2jZX2EghfzY4oG5kpW8NgUhgAZ6Rhqt5PjrGbS4YdM`

on beta.kodadot.xyz works fine, that's `main` branch.

image

**Comment by yangwao** [comment]
I highly sense there will be a lot of events and probably writing rubick resolver would be needed. 
wyt @vikiival 👀

**Comment by yangwao** [comment]
Okay, let me know when it's ready for testing :D 

image

**Comment by vikiival** [comment]
> [V]


pls use `[X]`

**Comment by vikiival** [comment]
> I highly sense there will be a lot of events and probably writing rubick resolver would be needed.

Yup the query is not optimal

**Comment by MSghais** [comment]
Hi @yangwao | KodaDot#1123 thanks ! 

1. For the substrate address format, I didn't change the address format for the profil.
Personaly,  when I click on the profil, I go directly on KSM ss58 address profil, and the history load very fastly.

I look sometimes with have some issues with the KSM format address on Profil.

We have already see this issues created for the KSM format address, it's the PR #2857. 
I take a look on it probably after please.

**Comment by vikiival** [comment]
killian can u please check it? 🥺 🙏

**Comment by yangwao** [comment]
Seems can't test on my end 
https://deploy-preview-2973--koda-nuxt.netlify.app/rmrk/u/CuHWHNcBt3ASMVSJmcJyiBWGxxiWLyjYoYbGjfhL4ovoeSd?tab=history

```
{"errors":[{"message":"Field \"limit\" is not defined by type \"EventWhereInput\".","extensions":{"code":"GRAPHQL_VALIDATION_FAILED","exception":{"stacktrace":["GraphQLError: Field \"limit\" is not defined by type \"EventWhereInput\".","    at ObjectField (/squid/node_modules/graphql/validation/rules/ValuesOfCorrectTypeRule.js:73:29)","    at Object.enter (/squid/node_modules/graphql/language/visitor.js:323:29)","    at Object.enter (/squid/node_modules/graphql/utilities/TypeInfo.js:370:25)","    at visit (/squid/node_modules/graphql/language/visitor.js:243:26)","    at validate (/squid/node_modules/graphql/validation/validate.js:69:24)","    at validate (/squid/node_modules/apollo-server-core/dist/requestPipeline.js:186:39)","    at processGraphQLRequest (/squid/node_modules/apollo-server-core/dist/requestPipeline.js:98:34)","    at runMicrotasks (<anonymous>)","    at processTicksAndRejections (node:internal/process/task_queues:96:5)","    at async processHTTPRequest (/squid/node_modules/apollo-server-core/dist/runHttpQuery.js:186:30)"]}}},{"message":"Field \"offset\" is not defined by type \"EventWhereInput\".","extensions":{"code":"GRAPHQL_VALIDATION_FAILED","exception":{"stacktrace":["GraphQLError: Field \"offset\" is not defined by type \"EventWhereInput\".","    at ObjectField (/squid/node_modules/graphql/validation/rules/ValuesOfCorrectTypeRule.js:73:29)","    at Object.enter (/squid/node_modules/graphql/language/visitor.js:323:29)","    at Object.enter (/squid/node_modules/graphql/utilities/TypeInfo.js:370:25)","    at visit (/squid/node_modules/graphql/language/visitor.js:243:26)","    at validate (/squid/node_modules/graphql/validation/validate.js:69:24)","    at validate (/squid/node_modules/apollo-server-core/dist/requestPipeline.js:186:39)","    at processGraphQLRequest (/squid/node_modules/apollo-server-core/dist/requestPipeline.js:98:34)","    at runMicrotasks (<anonymous>)","    at processTicksAndRejections (node:internal/process/task_queues:96:5)","    at async processHTTPRequest (/squid/node_modules/apollo-server-core/dist/runHttpQuery.js:186:30)"]}}}]}
```

image

**Comment by kkukelka** [comment]
> killian can u please check it? 🥺 🙏

interesting... it actually loads quite fast for me initially and then on like page 8 i get same state like wao

Screenshot 2022-05-09 at 19 43 50

**Comment by kkukelka** [comment]
Screenshot 2022-05-09 at 19 45 06

File: components/rmrk/Profile/ProfileDetail.vue [hunk 1]
```diff
@@ -438,11 +482,112 @@ export default class Profile extends mixins(PrefixMixin, InfiniteScrollMixin) {
...
+    try {
+      const { data } = await this.$apollo.query<{ events: Interaction[] }>({
+        query: allEventsByProfil,
+        client: 'subsquid',
+        variables: {
+          id: this.id,
+          search: {
+            caller_eq: this.id,
+            // first: this.first,
+            // offset: (this.currentValue - 1) * this.first,
```
Comment (kkukelka) [inline components/rmrk/Profile/ProfileDetail.vue:531]: maybe remove this if not used?

File: components/rmrk/Profile/ProfileDetail.vue [hunk 2]
```diff
@@ -438,11 +482,112 @@ export default class Profile extends mixins(PrefixMixin, InfiniteScrollMixin) {
...
+          },
+        },
+      })
+      if (data && data.events && data.events.length) {
+        let events: Interaction[] = data.events
+        // TODO : default value of HISTORY for BUY
```
Comment (kkukelka) [inline components/rmrk/Profile/ProfileDetail.vue:537]: is this still being done in the current PR?

File: components/rmrk/Profile/ProfileDetail.vue [hunk 3]
```diff
@@ -266,9 +300,11 @@ export default class Profile extends mixins(PrefixMixin, InfiniteScrollMixin) {
   readonly nftListByIssuer = nftListByIssuer
   readonly nftListCollected = nftListCollected
   readonly nftListSold = nftListSold
+  private openHistory = true
 
   public async mounted() {
     await this.fetchProfile()
+    // this.fetchCollectionEvents()
```
Comment (kkukelka) [inline components/rmrk/Profile/ProfileDetail.vue:307]: remove if not needed

**Review by kkukelka — CHANGES_REQUESTED** [review]

**Review by MSghais — COMMENTED** [review]
Unused vars deleted @kkukelka, and the code self reviewed too. Let me know please !

**Comment by kkukelka** [comment]
> Unused vars deleted @kkukelka, and the code self reviewed too. Let me know please !

sweet, could you take a look though at the weird bug me and wao experienced when clicking through pagination?

**Comment by MSghais** [comment]
> > Unused vars deleted @kkukelka, and the code self reviewed too. Let me know please !
> 
> sweet, could you take a look though at the weird bug me and wao experienced when clicking through pagination?

Hum I take a look on it, but I don't have the bug me too on this page.
Certainly a problem of pagination on History component, I don't touch the pagination on this PR.
@kkukelka 

image

**Comment by kkukelka** [comment]
> > > Unused vars deleted @kkukelka, and the code self reviewed too. Let me know please !
> > 
> > 
> > sweet, could you take a look though at the weird bug me and wao experienced when clicking through pagination?
> 
> Hum I take a look on it, but I don't have the bug me too on this page. Certainly a problem of pagination on History component, I don't touch the pagination on this PR. @kkukelka
> 
> image

alright then. created a followup ticket if someone wants to dig deeper here
https://github.com/kodadot/nft-gallery/issues/2997

**Review by kkukelka — APPROVED** [review]

**Comment by yangwao** [comment]
Technically it's on 9 page, so when you are going from 8 -> 9 , 10 -> 11, 21 -> 22
yet it stuck render on prev page.

https://deploy-preview-2973--koda-nuxt.netlify.app/rmrk/u/CuHWHNcBt3ASMVSJmcJyiBWGxxiWLyjYoYbGjfhL4ovoeSd?tab=history&page=9

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 88.18 USD/KSM ~ 1.134 $KSM
🧗 [EFuhGSJpetTgoZ2dfWxwirxADBWgAzWmVfJmTCauZuPxz6z](https://kusama.subscan.io/account/EFuhGSJpetTgoZ2dfWxwirxADBWgAzWmVfJmTCauZuPxz6z)
🔗 [0x836c9cbb8de40763c2528b99ae83a653faa59b0a7ff009854d7a150a8231d436](https://kusama.subscan.io/extrinsic/0x836c9cbb8de40763c2528b99ae83a653faa59b0a7ff009854d7a150a8231d436)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**